pkg/apis/resource/validation/validation.go

```diff
@@ -312,10 +331,14 @@ func validateDeviceConstraint(constraint resource.DeviceConstraint, fldPath *fie
 			return validateRequestNameRef(name, fldPath, requestNames)
 		},
 		stringKey, fldPath.Child("requests"))...)
-	if constraint.MatchAttribute == nil {
-		allErrs = append(allErrs, field.Required(fldPath.Child("matchAttribute"), ""))
-	} else {
+	if constraint.MatchAttribute != nil {
 		allErrs = append(allErrs, validateFullyQualifiedName(*constraint.MatchAttribute, fldPath.Child("matchAttribute"))...)
+	} else if constraint.DistinctAttribute != nil {
+		allErrs = append(allErrs, validateFullyQualifiedName(*constraint.DistinctAttribute, fldPath.Child("distinctAttribute"))...)
+	} else if utilfeature.DefaultFeatureGate.Enabled(features.DRAConsumableCapacity) {
+		allErrs = append(allErrs, field.Required(fldPath, "require either one of `matchAttribute`, `distinctAttribute`"))
```

Comment:
I think we can use similar language as for other one-offs here: https://github.com/kubernetes/kubernetes/blob/894ad4ec6315305ee59bb2c0a3b5eac0a1aee21b/pkg/apis/resource/validation/validation.go#L196C54-L196C110